python/lsst/pipe/tasks/multiBand.py

```diff
@@ -1123,7 +1368,7 @@ def readSources(self, dataRef):
         We also need to add columns to hold the measurements we're about to make
         so we can measure in-place.
         """
-        merged = dataRef.get(self.config.coaddName + "Coadd_mergeDet", immediate=True)
+        merged = dataRef.get(self.config.coaddName + self.inputCatalog, immediate=True)
```

Comment:
This will produce different dataset names, is that ok? There is no Coadd_ part of the string in your new version.

Reply:
This is expected. Measurements can now be run on one of three different catalogs: the flux conserved deblender output `Coadd_deblendedFlux`, the scarlet model output `Coadd_deblendedModel`  or the merged catalog (if no deblender is run) `Coadd_mergeDet`.